### PR TITLE
Improve detection of root block device for dracut

### LINKED ISSUE #112: Fedora 26 does not persist through reboot
4.12.13-300.fc26.x86_64

Unsure if this is related to #93, but `/boot/<machine-id>` was removed before installing. `dattobd`, `etc/datto/dla`, and `usr/sbin/datto_reload` are in initramfs.

The module is loaded during boot, but the reload scripts are not run.

### PR DESCRIPTION
**Pull request by crawfxrd**
Fixes #112

Need to test with all dracut systems. 😩

### CODE REVIEW

File: dist/initramfs/dracut/dattobd.sh [hunk 1]
```diff
@@ -1,52 +1,62 @@
...
-fi
+[ -z "$root" ] && root=$(getarg root=)
+[ -z "$rootfstype" ] && rootfstype=$(getarg rootfstype=)
+
+rbd="${root#block:}"
+if [ -n "$rbd" ]; then
+    # Based on 98dracut-systemd/rootfs-generator.sh
+    case "$rbd" in
+        LABEL=*)
+            rbd="$(echo  $rbd | sed 's,/,\\x2f,g')"
```
Comment (nixomose) [inline dist/initramfs/dracut/dattobd.sh]: what does this buy you? who needs the escaping?

Comment (nixomose) [reply dist/initramfs/dracut/dattobd.sh]: I see, never mind

File: dist/initramfs/dracut/dattobd.sh [hunk 2]
```diff
@@ -1,52 +1,62 @@
...
+    _info "root block device = $rbd"
 
+    # Device might not be ready
+    if ! [ -b "$rbd" ]; then
+        udevadm settle
+    fi
+
+    # Kernel cmdline might not specify rootfstype
+    #[ -z "$rootfstype" ] && rootfstype=$(blkid -s TYPE "$rbd" -o value)
+    [ -z "$rootfstype" ] && rootfstype="auto"
```
Comment (nixomose) [inline dist/initramfs/dracut/dattobd.sh]: auto always works? even for older dracutted oses?

Comment (crawfxrd) [reply dist/initramfs/dracut/dattobd.sh]: Will have to test. I expect I'll switch it back to the `blkid` one after trying, since it's worked so far.

**Pull request merged**